astropy/io/ascii/ui.py

```diff
@@ -220,7 +220,7 @@ def _guess(table, read_kwargs, format, fast_reader):
             reader.guessing = True
             return reader.read(table)
 
-        except (core.InconsistentTableError, ValueError, TypeError,
+        except (core.InconsistentTableError, ValueError, TypeError, AttributeError,
```

Comment:
I've actually thought about catching this and raising an `InconsistentTableError` for it. While we can think about it for this specific case, I think it is better go with a broader approach:
We don't know know what readers we might have in the future and even in the existing readers there might be many edge cases like this one not yet discovered (e.g. we obviously did not think that the reader could possibly fail this way when we wrote it otherwise that error would have been caught).

Thus, I considered - instead of listing selected exceptions - to just catch *all* exceptions here. After all, we are *guessing* and if the guess process does not find a way to read the table it will say "they using guessing=False and see what goes wrong". However, I realized that I would probably not get that approach through your review, and thus opted for the middle ground. I think the most likely exceptions (those defined in `io.ascii` plus Value, Type and AttributeError) are now covered.

On second thought I want to advertise **catching all exceptions**. That's future proof, low maintenance and easy to understand.
If we want to catch certain exceptions and re-raise them as `InconsistentTableError` that's fine in case the reader is selected explicitly.